ui/app/ducks/alerts/unconnected-account.js

```diff
@@ -0,0 +1,101 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { captureException } from '@sentry/browser'
+
+import actionConstants from '../../store/actionConstants'
+import { addPermittedAccount } from '../../store/actions'
+import { getOriginOfCurrentTab } from '../../selectors/selectors'
+
+// Constants
+
+export const ALERT_STATE = {
+  CLOSED: 'CLOSED',
+  ERROR: 'ERROR',
+  LOADING: 'LOADING',
+  OPEN: 'OPEN',
+}
+
+const name = 'unconnectedAccount'
+
+const initialState = {
+  state: ALERT_STATE.CLOSED,
+  address: null,
+}
+
+// Slice (reducer plus auto-generated actions and action creators)
+
+const slice = createSlice({
+  name,
+  initialState,
+  reducers: {
+    connectAccountFailed: (state) => {
+      state.state = ALERT_STATE.ERROR
+    },
+    connectAccountRequested: (state) => {
+      state.state = ALERT_STATE.LOADING
+    },
+    connectAccountSucceeded: (state) => {
+      state.state = ALERT_STATE.CLOSED
+      state.address = null
+    },
+    dismissAlert: (state) => {
+      state.state = ALERT_STATE.CLOSED
+      state.address = null
+    },
+    switchedToUnconnectedAccount: (state, action) => {
+      state.state = ALERT_STATE.OPEN
+      state.address = action.payload
+    },
+  },
+  extraReducers: {
```

Comment:
This doesn't need to block review, but I'm working on an improvement to this reducer that would remove the necessity to cache the current selected address. This approach of caching the relevant background state and watching for state updates doesn't scale well when you want to watch a larger set of things, and keeping redundant state in reducers is generally not advised. This was particularly noticeable while I was working on the next alert, which also needs to watch for changes to the current active tab.

Reply:
This has been completed in bf55bbaee0f75d0230a16525cc6073a887dbc65b